src/transformers/models/autoformer/modeling_autoformer.py

```diff
@@ -1570,20 +1572,23 @@ class AutoformerModel(AutoformerPreTrainedModel):
     def __init__(self, config: AutoformerConfig):
         super().__init__(config)
 
-        if config.scaling:
+        if config.scaling == "mean" or config.scaling:
             self.scaler = MeanScaler(dim=1, keepdim=True)
+        elif config.scaling == "std":
+            self.scaler = StdScaler(dim=1, keepdim=True)
         else:
             self.scaler = NOPScaler(dim=1, keepdim=True)
 
-        self.embedder = FeatureEmbedder(
-            cardinalities=config.cardinality,
-            embedding_dims=config.embedding_dimension,
-        )
+        if config.num_static_categorical_features > 0:
+            self.embedder = FeatureEmbedder(
+                cardinalities=config.cardinality,
+                embedding_dims=config.embedding_dimension,
+            )
 
         # transformer encoder-decoder and mask initializer
         self.encoder = AutoformerEncoder(config)
         self.decoder = AutoformerDecoder(config)
-        self.decomposition_layer = AutoformerSeriesDecompositionLayer(config.moving_avg)
```

Comment:
revert this delete after Informer merge

Reply:
added in https://github.com/huggingface/transformers/pull/21891/commits/e61dac5b68baefdcbd339de545041e55ae535131